### PR TITLE
Settings modal

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

**Comment by iqmulator** [comment]
image
image
image
image
image

File: src/dotnet/Chat.UI.Blazor/Components/Settings/NightModeToggle.razor [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+<Toggle Value="@_isDarkTheme" ValueChanged="OnThemeChanged" Class="toggle-sm"/>
+
+@code {
+    private bool _isDarkTheme;
+
+    [Inject] ThemeUI ThemeUI { get; init; } = null!;
+
+    private void OnThemeChanged()
+        => ThemeUI.CurrentTheme.Value = ThemeUI.CurrentTheme.ValueOrDefault == Theme.Dark ? Theme.Light : Theme.Dark;
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Settings/NightModeToggle.razor:11]: Can you clarify this trick?

File: src/dotnet/UI.Blazor/Components/SettingsPanel/SettingsTab.razor [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    <div class="settings-tab-header">
+        <SettingsHeader Title="@Title" OnArrowClick="@OnChangeView"/>
+    </div>
+    <div class="settings-tab-content">
+        @ChildContent
+    </div>
+</div>
+
+@code {
+    [CascadingParameter] public SettingsPanel Panel { get; set; } = null!;
```
Comment (FrolYo) [inline src/dotnet/UI.Blazor/Components/SettingsPanel/SettingsTab.razor:19]: Passing Panel as a cascading parameter looks very weird

File: src/dotnet/Chat.UI.Blazor/Module/ChatBlazorUIModule.cs [hunk 3]
```diff
@@ -53,5 +54,19 @@ public override void InjectServices(IServiceCollection services)
         services.AddScoped<ChatActivity>();
         services.AddScoped<UnreadMessagesFactory>();
         fusion.AddComputeService<ChatRecordingActivity>(ServiceLifetime.Transient);
+
+        services.ConfigureLifetimeEvents(events =>
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Module/ChatBlazorUIModule.cs:58]: what kind of issue is solved with such an approach?

**Review by alexis-kochetov — APPROVED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/Settings/NightModeToggle.razor (hunk 1 above)
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/Settings/NightModeToggle.razor:11]: I mean that value from input is not used - you are always trying to invert value stored in ThemeUI

**Pull request merged**